src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoderEncoderViewModel.cs

```diff
@@ -0,0 +1,653 @@
+            return new ECDsaSecurityKey(ecdSa);
+        }
+
+        private AsymmetricKeyParameter? GetPrivateAsymmetricKeyParameter()
+        {
+            if (string.IsNullOrWhiteSpace(PrivateKey))
+            {
+                JwtValidation.IsValid = false;
+                JwtValidation.ErrorMessage = LocalizedStrings.InvalidPrivateKeyError;
+                return null;
+            }
+            var privateKeyStringBuilder = new StringBuilder(PrivateKey!.Trim());
+            if (!PrivateKey!.StartsWith(PrivateKeyStart))
+            {
+                privateKeyStringBuilder.Insert(0, PrivateKeyStart);
+            }
+            if (!PrivateKey.EndsWith(PrivateKeyEnd))
+            {
+                privateKeyStringBuilder.Append(PrivateKeyEnd);
+            }
+
+            var pemReader = new PemReader(new StringReader(privateKeyStringBuilder.ToString()));
+            object? pemObject = pemReader.ReadObject();
+            if (pemObject is null)
+            {
+                JwtValidation.IsValid = false;
+                JwtValidation.ErrorMessage = LocalizedStrings.InvalidPrivateKeyError;
+                return null;
+            }
+
+            if (pemObject is AsymmetricKeyParameter)
+            {
+                return pemObject as AsymmetricKeyParameter;
+            }
+            else if (pemObject is AsymmetricCipherKeyPair)
+            {
+                var pair = pemObject as AsymmetricCipherKeyPair;
+                return pair!.Private;
+            }
+
+            JwtValidation.IsValid = false;
+            JwtValidation.ErrorMessage = LocalizedStrings.InvalidPrivateKeyError;
+            return null;
+        }
+
+        /// <summary>
+        /// Source (https://stackoverflow.com/questions/28370414/import-rsa-key-from-bouncycastle-sometimes-throws-bad-data)
+        /// </summary>
+        private static byte[] ConvertRSAParametersField(BigInteger n, int size)
+        {
+            byte[] bs = n.ToByteArrayUnsigned();
+            if (bs.Length == size)
+            {
+                return bs;
+            }
+            if (bs.Length > size)
+            {
+                throw new ArgumentException("Specified size too small", "size");
+            }
+            byte[] padded = new byte[size];
+            Array.Copy(bs, 0, padded, size - bs.Length, bs.Length);
+            return padded;
+        }
+
+        #endregion
+
+        private void IsSignatureRequired(JwtAlgorithmDisplayPair value)
+        {
+            if (value.Value is JwtAlgorithm.HS256 ||
+                value.Value is JwtAlgorithm.HS384 ||
+                value.Value is JwtAlgorithm.HS512)
+            {
+                RequireSignature = true;
+            }
+            else
+            {
+                RequireSignature = false;
+            }
+        }
+
+        protected override void TreatComputationQueue(bool value)
+        {
+            throw new NotImplementedException();
```

Comment:
Since we're not implementing `QueueWorkerViewModelBase`, should we remove this inheritance?